source/game.broadcast.genredefinition.movie.bmx

```diff
@@ -71,14 +73,162 @@ Type TMovieGenreDefinitionCollection
+		Local all:TMovieGenreDefinition[] = new TMovieGenreDefinition[ids.length]
+		Local i:Int
+		Local floats:Float[ids.length]
+		For i:Int = 0 Until ids.length
+			all[i] = Get([ids[i]])
+		Next
+		Local main:TMovieGenreDefinition = all[0]
+		result.BadFollower = main.BadFollower
+		result.GoodFollower = main.GoodFollower
+		'TODO this is not yet optimal
+		'popularity is based on the main popularity
+		'creating and consistently maintaining an aggregated popularity 
+		'would be an enormous effort
+		result.referenceID = main.referenceID
+		result._popularity = main.GetPopularity()
+
+		result.castAttributes = CreateMap()
+		result.focusPointPriorities = CreateMap()
+		For i:Int = 0 Until ids.length
+			collectKeys(all[i].castAttributes, result.castAttributes)
+			collectKeys(all[i].focusPointPriorities, result.focusPointPriorities)
+		Next
+
+		'audience attraction
+		result.AudienceAttraction = main.AudienceAttraction.Copy()
+		Local att:TAudience = result.AudienceAttraction
+
+		Select all.length
+			Case 1
+				addWeighted(main.castAttributes, result.castAttributes, 1.0, 0.0)
+
+				addWeighted(main.focusPointPriorities, result.focusPointPriorities, 1.0, 1.0)
+				
+				'attraction already copied
+			Case 2
+				addWeighted(main.castAttributes, result.castAttributes, 0.75, 0.0)
+				addWeighted(all[1].castAttributes, result.castAttributes, 0.25, 0.0)
+
+				addWeighted(main.focusPointPriorities, result.focusPointPriorities, 0.75, 1.0)
+				addWeighted(all[1].focusPointPriorities, result.focusPointPriorities, 0.25, 1.0)
+
+				att.Multiply(0.75).add(attr(all[1],0.25))
+			Case 3
+				addWeighted(main.castAttributes, result.castAttributes, 0.6, 0.0)
+				addWeighted(all[1].castAttributes, result.castAttributes, 0.25, 0.0)
+				addWeighted(all[2].castAttributes, result.castAttributes, 0.15, 0.0)
+
+				addWeighted(main.focusPointPriorities, result.focusPointPriorities, 0.6, 1.0)
+				addWeighted(all[1].focusPointPriorities, result.focusPointPriorities, 0.25, 1.0)
+				addWeighted(all[2].focusPointPriorities, result.focusPointPriorities, 0.15, 1.0)
+
+				att.Multiply(0.6).add(attr(all[1],0.25)).add(attr(all[2],0.15))
+			Case 4
+				addWeighted(main.castAttributes, result.castAttributes, 0.6, 0.0)
+				addWeighted(all[1].castAttributes, result.castAttributes, 0.20, 0.0)
+				addWeighted(all[2].castAttributes, result.castAttributes, 0.12, 0.0)
+				addWeighted(all[3].castAttributes, result.castAttributes, 0.08, 0.0)
+
+				addWeighted(main.focusPointPriorities, result.focusPointPriorities, 0.6, 1.0)
+				addWeighted(all[1].focusPointPriorities, result.focusPointPriorities, 0.20, 1.0)
+				addWeighted(all[2].focusPointPriorities, result.focusPointPriorities, 0.12, 1.0)
+				addWeighted(all[3].focusPointPriorities, result.focusPointPriorities, 0.08, 1.0)
+
+				att.Multiply(0.6).add(attr(all[1],0.20)).add(attr(all[2],0.12)).add(attr(all[2],0.08))
+			Default throw "too many genres "+ all.length
```

Comment:
Koennte man nicht einfach sagen "ab Case 4" wird jedes weitere Genre ignoriert? Also einfach "Case 4" in "default" umbenennen und dort dann nur die ersten 4 behandeln?

Reply:
Ich werde das noch anpassen.